osm2streets/src/transform/intersection_geometry.rs

```diff
@@ -49,21 +54,29 @@ pub fn generate(streets: &mut StreetNetwork, timer: &mut Timer) {
                         road.center_line.last_pt()
                     };
                     set_polygons.push((i.id, Circle::new(pt, Distance::meters(3.0)).to_polygon()));
-
-                    // Also don't attempt to make Movements later!
-                    make_stop_signs.push(i.id);
                 } else {
                     remove_dangling_nodes.push(i.id);
                 }
             }
         }
     }
+
+    for road in streets.roads.values_mut() {
+        let untrimmed = road.get_untrimmed_center_line(streets.config.driving_side);
+        if let Some(pl) =
+            Road::trim_polyline_both_ends(untrimmed.clone(), road.trim_start, road.trim_end)
+        {
+            road.center_line = pl;
+        } else {
+            // TODO Mark for collapsing?
+            error!("{} got trimmed into oblivion", road.id);
```

Comment:
This shows up consistently where I expect it to / where the road looks visibly untrimmed. Maybe it's as simple as setting `internal_junction_road` and repeating a few transformation passes. That's a little unsatisfying in the long run... I wouldn't want to wind up regenerating geometry in the whole network after fixing a few things.